bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java

```diff
@@ -3193,7 +3193,7 @@ private Rectangle translateRectangleInPixelsInDisplayCoordinateSystem(int x, int
 
 private Rectangle translateRectangleInPixelsInDisplayCoordinateSystem(int x, int y, int width, int height, Monitor monitorOfLocation, Monitor monitorOfArea) {
 	Point topLeft = getPixelsFromPoint(monitorOfLocation, x, y);
-	int zoom = DPIUtil.getZoomForAutoscaleProperty(monitorOfArea.zoom);
+	int zoom = isRescalingAtRuntime() ? DPIUtil.getZoomForAutoscaleProperty(monitorOfArea.zoom) : getDeviceZoom();
```

Comment:
It is a one liner, because of the special notation we can use. here. But this logic must be applied the same on all places and it is some kind of logic, so I would vote for that.

Reply:
done